proto/directions.proto

```diff
@@ -0,0 +1,94 @@
+package valhalla;
+
+message Waypoint {
+  optional float lat = 1;
+  optional float lon = 2;
+  optional string type = 3;
```

Comment:
my instinct here is that we know what type it is based on whether we are returning multiple legs or not - the waypoints at the beginning/end of the legs are fulls tops and the others are through-points. Open to discussion, though

Reply:
if you have `through` waypoints - that does not create multiple legs

Reply:
exactly, so if you have one leg, you can assume that any waypoints that aren't the start/end points are `through` waypoints

Reply:
how do you know what they are for more than a single leg? break,through,through,break,break,through,break

Reply:
any waypoint that is a start/end point of a leg is a break, all others are through. Am I missing something?

Reply:
it is a single list of waypoints - so information is not duplicated - it is not per leg

Reply:
okay, I was just thinking that you could imply this info from the values in the legs, but I'm fine with keeping it.